ddtrace/debugging/_encoding.py

```diff
@@ -65,7 +65,7 @@ def put(self, item):
             raise BufferFull(self.size, size)
 
         if self.size > 2:
-            size += 1
+            self.size += 1
```

Comment:
do we need a lock around this `put` operation?


assuming I understand the issue... race condition between increasing the buffer size and not incrementing the size until later?